### PR TITLE
use auto & cell instead of auto cell

### PR DESCRIPTION
**Pull request by tamiko**
Unify the range-based for loops over active cells as discussed in #8102

### CODE REVIEW

File: examples/step-49/doc/results.dox [hunk 1]
```diff
@@ -271,7 +271,7 @@ Triangulation<3> triangulation;
 create_3d_grid(triangulation);
 triangulation.set_manifold(cylinder_id, cylinder);
 
-for (auto cell : triangulation.active_cell_iterators())
+for (auto & cell : triangulation.active_cell_iterators())
```
Comment (kronbichler) [inline examples/step-49/doc/results.dox]: Is there a reason why clang-format sometimes puts a space between `& cell` and sometimes not?

**Comment by luca-heltai** [comment]
Wouldn't `const auto &cell` be even better?

File: examples/step-49/doc/results.dox (hunk 1 above)
Comment (bangerth) [reply examples/step-49/doc/results.dox]: This is in a .dox file. clang-format doesn't touch those. I suspect that @tamiko used the version with the space in his regex-replace, and then clang-format removed the space in all places it gets to see.

File: include/deal.II/differentiation/ad/ad_helpers.h [hunk 2]
```diff
@@ -1081,7 +1081,7 @@ namespace Differentiation
      *   Vector<double> cell_rhs (...);
      *
      *   // Assembly loop:
-     *   for (auto cell & : ...)
+     *   for (auto & cell : ...)
```
Comment (bangerth) [inline include/deal.II/differentiation/ad/ad_helpers.h]: ha!

**Comment by bangerth** [comment]
On 5/12/19 2:27 AM, Luca Heltai wrote:
> Wouldn't |const auto &cell| be even better?

That's what I have been writing for the most part.

File: include/deal.II/differentiation/ad/ad_helpers.h (hunk 2 above)
Comment (jppelteret) [reply include/deal.II/differentiation/ad/ad_helpers.h]: whoops!

File: examples/step-49/doc/results.dox (hunk 1 above)
Comment (tamiko) [reply examples/step-49/doc/results.dox]: Exactly

**Comment by tamiko** [comment]
I have updated this PR (fixing the code snippets in the dox files). I have removed the change in `dof_handler_policy.cc` which will hopefully make the CI happy.

**Pull request merged**